NPM-only components

notlee opened the pull request:
This is the beta branch. Changes were reviewed here: https://github.com/Financial-Times/create-origami-component/pull/235 (and elsewhere)

notlee committed:
Create an Origami Component following v2 of the specification

Draft Specification: https://github.com/Financial-Times/origami-website/pull/273

Closes: https://github.com/Financial-Times/create-origami-component/issues/220

notlee commented:
Percy shows the icons are missing because the image service hasn't released yet

chee committed:
Move description and keywords to the package.json

chee committed:
Update to current impl of workflows

JakeChampion commented on templates/npmignore.js:
```diff
@@ -0,0 +1,18 @@
+.idea/
+/demos/local
+/coverage
+.eslintrc.js
+.github
+.gitignore
+.remarkrc.js
+.stylelintrc.js
```
```suggestion
.eslintrc.cjs
.github/
.gitignore
.remarkrc.cjs
.stylelintrc.cjs
```

JakeChampion commented on templates/origami-manifest.js:
```diff
@@ -1,10 +1,8 @@
 	"origamiType": "module",
 	"origamiCategory": "${answers.category}",
-	"origamiVersion": 1,
+	"origamiVersion": 2,
```
```suggestion
	"origamiVersion": "2.0",
```

JakeChampion commented on templates/package-json.js:
```diff
@@ -1,16 +1,69 @@
+	};
+
+	if (answers.javascript) {
+		Object.assign(packageJson, {
+			"exports": {
+				".": {
+					"browser": "./main.js"
+				}
+			},
+			"browser": "./main.js",
+			"types": "./types/main.d.ts"
+		});
```
Let's remove the typescript parts for now, we haven't added it to any of the spec v2 components we are currently working on
```suggestion
		Object.assign(packageJson, {
			"browser": "./main.js",
		});
```

JakeChampion commented on templates/package-json.js:
```diff
@@ -1,16 +1,69 @@
+			"types": "./types/main.d.ts"
+		});
+
+		Object.assign(packageJson.scripts, {
+			"build": "tsc src/**/*.js main.js --declaration --allowJs --emitDeclarationOnly --outDir types",
+			"version": "npm run build"
+		});
```
```suggestion
```

JakeChampion commented on templates/package-json.js:
```diff
@@ -1,16 +1,69 @@
+
+		Object.assign(packageJson.devDependencies, {
+			"eslint": "^7.19.0",
+			"eslint-config-origami-component": "^2.1.0",
+			"typescript": "^4.0.5"
```
```suggestion
			"eslint-config-origami-component": "^2.1.0"
```

JakeChampion committed:
make npm ignore the new extension being used for the linting config files

JakeChampion committed:
correct the origami version to be a string

JakeChampion committed:
Remove typescript config from the component for now

JakeChampion committed:
update the repository links to the new location

JakeChampion committed:
fix indent issue in percy workflow

JakeChampion committed:
update tests to check for the new workflows added

JakeChampion committed:
make keywords be an array and not a string

JakeChampion committed:
Create the package.json in the exact same format that npm would

Without this change, running `npm install` on a just generated component would make npm modify the package.json. It would re-order the devDependencies into alphabetical order and remove the dependencies object

JakeChampion committed:
make the browser field be the exact value the specification states: "main.js"

JakeChampion committed:
Remove support for node 10

JakeChampion committed:
Remove index.scss as we are reserving this file for future sass changes

JakeChampion committed:
fix the o-brand import

we do not have _index.scss files so need to import the exact file name now

JakeChampion committed:
add package.json to list of files which trigger a percy test

JakeChampion committed:
use the prerelease of obt

JakeChampion committed:
update workflows to the currently used versions in our 2.0 components

JakeChampion committed:
update package-lock.json

JakeChampion committed:
use https instead of ssh for github dependencies

notlee committed:
Drop "module" type for "component" for spec v2 components

Conversation: https://github.com/Financial-Times/origami-website/pull/273#discussion_r617396987

JakeChampion committed:
run npm-test workflow on all branches and pull-requests (#263)

notlee commented:
This released as `v0.3.2` since I used the patch release label but the existing beta tag does not reference this branch.

notlee committed:
Update package-lock.json

Using npm 6 as we do not require 7

notlee committed:
Remove "origamiCategory" from boilerplate.

https://github.com/Financial-Times/origami/pull/120

notlee committed:
Remove silent mode.

- It has been deprecated for some time.
- Build Service v3 no longer uses it.

notlee committed:
Remove links and references to the now deprecated Origami spec.

Links to the Origami website 404 currently. Depends on:
Financial-Times/origami-website#361

JakeChampion commented on package.json:
```diff
@@ -19,7 +19,7 @@
     "mocha": "^8.0.1",
     "nixt": "github:vesln/nixt#d2075099ecf935c870705fa5d5de437113d4d272",
-    "origami-build-tools": "^10.5.9",
+    "origami-build-tools": "prerelease",
```
```suggestion
    "origami-build-tools": "^11",
```

JakeChampion replied:
fixed in [`7733640` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/7733640bb2ff9c761fd2babfa3fe5e8260d1db84)

JakeChampion commented on templates/package-json.js:
```diff
@@ -1,16 +1,67 @@
+		"devDependencies": {
+			"remark": "^12.0.1",
+			"remark-lint": "^7.0.1",
+			"remark-preset-lint-origami-component": "^2.0.0-beta.1"
```
```suggestion
			"remark-preset-lint-origami-component": "^2.0.0"
```

JakeChampion replied:
fixed in [`1b0e344` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/1b0e3449b950710fc2defeaab53eac4faeb36ea9)

JakeChampion commented on templates/package-json.js:
```diff
@@ -1,16 +1,67 @@
+
+	if (answers.brands) {
+		packageJson.peerDependencies = {
+			"@financial-times/o-brand": "^4.0.0-beta.0"
```
```suggestion
			"@financial-times/o-brand": "^4.0.0"
```

JakeChampion replied:
fixed in [`1b0e344` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/1b0e3449b950710fc2defeaab53eac4faeb36ea9)

JakeChampion commented on templates/workflows/test-origami-component.yml:
```diff
@@ -5,18 +5,11 @@ jobs:
-        GITHUB_TOKEN: ${{ secrets.ORIGAMI_FOX_ACCESS_TOKEN }}
-        ORIGAMI_PULL_REQUEST: ${{ github.event.pull_request.html_url }}
+    - run: npm install -g npm@^7
+    - run: npm install -g origami-build-tools@prerelease
```
```suggestion
    - run: npm install -g origami-build-tools@^11
```

JakeChampion replied:
fixed in [`7733640` (#238)](https://github.com/Financial-Times/create-origami-component/pull/238/commits/7733640bb2ff9c761fd2babfa3fe5e8260d1db84)

JakeChampion committed:
Make sure the repo is checked-out as the origami support user who can push back to github

JakeChampion committed:
use obt v11

JakeChampion committed:
use latest origami dependencies in the template files

origamiserviceuser commented:
:tada: This PR is included in version v1.0.0 :tada:
